app/src/androidTest/java/com/epfl/drawyourpath/database/FirebaseDatabaseUtilsTest.kt

```diff
@@ -0,0 +1,284 @@
+            val runSubnode = mock(DataSnapshot::class.java)
+            val runs = listOf(mockRun((message.content as MessageContent.RunPath).run))
+            `when`(runSubnode.children).thenReturn(runs)
+            `when`(runSubnode.value).thenReturn(true)
+            runSubnode
+        } else {
+            mockSnapshot(null)
+        }
+        `when`(snapshot.child(FirebaseKeys.CHAT_MESSAGE_CONTENT_RUN)).thenReturn(run)
+
+        val txt = mockSnapshot(
+            if (message.content is MessageContent.Text) {
+                (message.content as MessageContent.Text).text
+            } else {
+                null
+            },
+        )
+        `when`(snapshot.child(FirebaseKeys.CHAT_MESSAGE_CONTENT_TEXT)).thenReturn(txt)
+
+        return snapshot
+    }
+
+    @Test
+    fun transformDailyGoalsReturnsEmptyListWithNullSnapshot() {
+        assertTrue(FirebaseDatabaseUtils.transformDailyGoals(null).isEmpty())
+    }
+
+    @Test
+    fun transformDailyGoalsReturnsEmptyListWhenNoChildren() {
+        val snapshot = mock(DataSnapshot::class.java)
+        assertThat(FirebaseDatabaseUtils.transformDailyGoals(snapshot).isEmpty(), `is`(true))
+    }
+
+    @Test
+    fun transformDailyGoalsReturnsGoals() {
+        val dailyGoals = listOf(
+            DailyGoal(12.0, 12.0, 12),
+            DailyGoal(10.0, 10.0, 10),
+        )
+
+        val dailyGoalsSnapshots = dailyGoals.map { mockDailyGoalSnapshot(it) }
+
+        val snapshot = mock(DataSnapshot::class.java)
+        `when`(snapshot.children).thenReturn(dailyGoalsSnapshots)
+
+        dailyGoals.zip(FirebaseDatabaseUtils.transformDailyGoals(snapshot)).forEach {
+            assertThat(it.first, `is`(it.second))
+        }
+    }
+
+    @Test
+    fun getKeysOfNullSnapshotIsEmpty() {
+        Assert.assertEquals(FirebaseDatabaseUtils.getKeys(null).size, 0)
+    }
+
+    @Test
+    fun getKeysReturnsSnapshotKeys() {
+        val snapshot = mock(DataSnapshot::class.java)
+        val child1 = mock(DataSnapshot::class.java)
+        val child2 = mock(DataSnapshot::class.java)
+
+        `when`(child1.key).thenReturn("child1")
+        `when`(child2.key).thenReturn("child2")
+        `when`(snapshot.children).thenReturn(listOf(child1, child2))
+
+        assertThat(FirebaseDatabaseUtils.getKeys(snapshot), `is`(listOf("child1", "child2")))
+    }
+
+    @Test
+    fun transformChatListOfNullSnapshotIsEmpty() {
+        Assert.assertEquals(FirebaseDatabaseUtils.transformChatList(null).size, 0)
+    }
+
+    @Test
+    fun transformChatListOfEmptySnapshotIsEmpty() {
+        val chatList = mock(DataSnapshot::class.java)
+        `when`(chatList.children).thenReturn(emptyList())
+        Assert.assertEquals(FirebaseDatabaseUtils.transformChatList(chatList).size, 0)
+    }
+
+    @Test
+    fun transformRunReturnsNullForNullSnapshot() {
+        Assert.assertEquals(FirebaseDatabaseUtils.transformRun(null), null)
+    }
+
+    @Test
+    fun transformRunReturnsExpectedRuns() {
```

Comment:
I think there is a typo in the name of this function, it should be transformRunsReturnExpectedRuns(..) since we transform a list of runs